CMakeLists.txt

```diff
@@ -149,11 +149,6 @@ set(NVIM_API_LEVEL 13)        # Bump this after any API/stdlib change.
 set(NVIM_API_LEVEL_COMPAT 0)  # Adjust this after a _breaking_ API change.
 set(NVIM_API_PRERELEASE true)
 
-# Build-type: RelWithDebInfo
-# /Og means something different in MSVC
-if(CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
-   set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -Og -g")
```

Comment:
We explicitly wanted to use `-Og` for RelWithDebInfo with compilers that support it. Calling the compiler with `-O2 -g -Og -g` is perfectly fine, as you stated, since the latter option takes precedence.

If any options should be removed, it should be the first set.

Reply:
We should remove `-O2 -g`?  But `-O2 -g` is a default for RelWithDebInfo and many projects choose `-O2` over `-Og`.

The RelWithDebInfo configuration is designed to strike a balance between performance and debugging. -Og skews heavily towards debugging at the cost of runtime performance, which contradicts the purpose of this configuration.

e.g. LLVM use `-O2` for RelWithDebInfo

> RelWithDebInfo
> 
> These builds are useful when debugging. They generate optimized binaries with debug information. CMakes default optimization level is -O2. This can be configured by setting the CMAKE_CXX_FLAGS_RELWITHDEBINFO variable on the CMake command line.

https://releases.llvm.org/11.0.1/docs/GettingStarted.html

RelWithDebInfo means release builds that contains debug info.

Reply:
> If any options should be removed, it should be the first set.

I'm not sure I agree. The `-Og` optimization option was initially introduced for the `CMAKE_C_FLAGS_DEV` cmake option which was explicitly for development. It was then silently removed. So this was not a conscious decision, or at least not mentioned in writing.

We now use this for `RelWithDebInfo`, which we use for our nightly releases. `-Og` is the same as `-O1`, so we're heavily kneecapping our nightly release performance. I'm not convinced the performance cost is worth any theoretical debugging information (or at the very least it would be good to first confirm we actually benefit from this extra debugging information).